ketcher-autotests/tests/Structure-Creating-&-Editing/Actions-With-Structures/Rectangle selection/rectangle-selection-tool.spec.ts

```diff
@@ -0,0 +1,165 @@
+    const objectSelection = 100;
+    const atomNumber = 5;
+    await openFileAndAddToCanvas('KET/two-benzene-with-atoms.ket', page);
+    await page.getByTestId('select-rectangle').click();
+    const point = await selectObjects(page, objectSelection, objectSelection);
+    await clickOnAtom(page, 'C', atomNumber);
+    await dragMouseTo(point.x + xDelta, point.y - yDelta, page);
+  });
+
+  test('Reaction components selection', async ({ page }) => {
+    //  Test case: EPMLSOPKET-1349
+    const atomNumber = 5;
+    const moveMouseCoordinatesY = 10;
+    const moveMouseCoordinatesX = 270;
+    await openFileAndAddToCanvas('Rxn-V2000/benzene-chain-reaction.rxn', page);
+    await page.getByTestId('select-rectangle').click();
+    const point = await getCoordinatesOfTheMiddleOfTheScreen(page);
+    await page.mouse.move(
+      point.x - moveMouseCoordinatesX,
+      point.y + moveMouseCoordinatesY,
+    );
+    await page.mouse.down();
+    await page.mouse.up();
+    await clickCanvas(page);
+
+    await page.mouse.move(point.x, point.y + atomNumber);
+    await page.mouse.down();
+    await page.mouse.up();
+    await clickCanvas(page);
+
+    await page.keyboard.down('Shift');
+    await page.mouse.click(
+      point.x - moveMouseCoordinatesX,
+      point.y + moveMouseCoordinatesY,
+    );
+    await page.mouse.click(point.x, point.y + atomNumber);
+    await page.keyboard.up('Shift');
+    await clickCanvas(page);
+
+    await page.keyboard.press(`${modifier}+KeyA`);
+  });
+
+  test('Reaction components dragging', async ({ page }) => {
+    //  Test case: EPMLSOPKET-1350
+    const objectSelection = 100;
+    const moveMouseCoordinatesY = 10;
+    await openFileAndAddToCanvas('Rxn-V2000/benzene-chain-reaction.rxn', page);
+    await page.getByTestId('select-rectangle').click();
+    await clickCanvas(page);
+    const point = await selectObjects(page, xMark, yMark);
+    await clickOnAtom(page, 'C', moveMouseCoordinatesY);
+    await dragMouseTo(point.x - objectSelection, point.y - yMark, page);
+  });
+
+  test('Fusing atoms together', async ({ page }) => {
+    //  Test case: EPMLSOPKET-1351
+    const atomNumber = 4;
+    const atomLabel = 9;
+    await openFileAndAddToCanvas('KET/two-benzene-with-atoms.ket', page);
+    await page.getByTestId('select-rectangle').click();
+    await clickOnAtom(page, 'C', atomNumber);
+    const atomPoint = await getAtomByIndex(page, { label: 'C' }, atomLabel);
+    await dragMouseTo(atomPoint.x, atomPoint.y, page);
+  });
+
+  test('Fusing bonds together', async ({ page }) => {
+    //  Test case: EPMLSOPKET-1351
+    const moveMouseCoordinatesY = 10;
+    const objectSelection = 50;
+    await drawBenzeneRing(page);
+    await selectNestedTool(page, BondTool.SINGLE_AROMATIC);
+    const coordinates = await getCoordinatesTopAtomOfBenzeneRing(page);
+    await page.mouse.click(coordinates.x + xDelta, coordinates.y - yDelta);
+    await page.getByTestId('select-rectangle').click();
```

Comment:
Suggestion – your test looks overloaded. If you perform multiple actions like click here, then there, then drag and so on, then create a function for this. 
This is not about code duplication, this is about meaning. Just imagine, how would you describe your test. You don't tell people:
1) Move mouse here
2) Then click in this coordinates
3) Then drag

You tell – select structure. Of course, you can skip this step, when your test is 5 lines long. But when you have a lot of clicks, drags, mouse moves in your tests it becomes hard to read. So ideally, your test should look like:
```
await selectRectangleSelection();
await selectBenzeneRing();
await moveBenzeneRing();
...
```

Reply:
done